src/plugins/profiling/server/routes/search_flamechart.ts

```diff
@@ -77,39 +74,39 @@ export function registerFlameChartSearchRoute(router: IRouter<DataRequestHandler
         const esClient = context.core.elasticsearch.client.asCurrentUser;
         const filter = newProjectTimeQuery(projectID!, timeFrom!, timeTo!);
 
-        // const resp = await getCountResponse(context, filter);
+        // Start with counting the results in the index down-sampled by 5^6.
+        // That is in the middle of our down-sampled indexes.
+        const pow6 = 6;
         const resp = await esClient.search({
-          index: 'profiling-events-1000',
+          index: 'profiling-events-5pow' + pow6,
           body: {
             query: filter,
             size: 0,
             track_total_hits: true,
           },
         });
+        const sampleCountFromPow6 = resp.body.hits.total.value as number;
 
-        const sampleCountFromSmallestTable = resp.body.hits.total.value as number;
+        console.log('sampleCountFromPow6', sampleCountFromPow6);
 
         const [eventsIndex, sampleRate, estimatedSampleCount] = getSampledTraceEventsIndex(
           sampleSize,
-          sampleCountFromSmallestTable
+          sampleCountFromPow6,
+          pow6
         );
 
         // eslint-disable-next-line no-console
-        console.log('Index', eventsIndex, sampleRate, estimatedSampleCount);
+        console.log('Index', eventsIndex, sampleRate, estimatedSampleCount, index);
 
         const resEvents = await esClient.search({
           index: eventsIndex,
           body: {
             size: 0,
-            query: {
-              function_score: {
-                query: filter,
-              },
-            },
+            query: filter,
             aggs: {
               sample: {
                 sampler: {
-                  shard_size: sampleSize,
+                  shard_size: 100000,
```

Comment:
How does this behaves when there's more than 1 shard serving the data?

Reply:
100k is the upper limit of entries for a given query. If I understand correctly, with more shards we have 100k / #shards entries. But maybe we should discuss this in a chat.

Reply:
I removed this completely.